lib/Web/Main.hs

```diff
@@ -4,18 +4,73 @@
 
 module Web.Main
   ( runServer
+  , RunServerException (..)
   ) where
 
 import Backend.Commands (runCommand)
 import Config
+import Control.Exception (Exception, throw)
+import Data.Bifunctor (Bifunctor(first))
+import Data.Either (partitionEithers)
 import Data.Proxy
 import Network.Wai.Handler.Warp (run)
+import Options.Applicative
 import Servant.Server (serve)
+import System.Environment (lookupEnv)
+import Text.Interpolation.Nyan
+import Text.Read (readEither)
 import Web.API (API)
 import Web.Server
 
+data ServerOptions = ServerOptions { serverPort :: Maybe Int }
+
+data RunServerException
+  = RunServerIncorrectPort Int
+  | RunServerNoPortSpecified
+  | RunServerEnvVarParseFail String
+  deriving stock (Eq)
+
+instance Show RunServerException where
+  show (RunServerIncorrectPort port) = "Port number should be greater than 0. Your port number is: " ++ show port
+  show RunServerNoPortSpecified = "No port was specified. It can be set using the environment variable \"COFFER_SERVER_PORT\" or the \"--port=$port_num\" command-line option."
```

Comment:
I think it is a good idea to have a default port. This way we minimize the effort needed to start the application

Reply:
We briefly discussed this and decided not to have a default port, buuuuut I do see your point. Let's do it separately though, I've created an issue for this: #131

Reply:
Oh, okay, I didn't know about the discussion. Then LGTM 👍